annotatedcsv/reader.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+// Package annotatedcsv provides a reader for annotated CSV sources.
+// Annotated CSV must contain at least an annotation with data types definition.
+// The first row after annotations must be a line with column names.
+//
+// Annotated CSV with multiple sections example:
+//		#datatype,string,long,dateTime:RFC3339,double,string,string
+//		#group,false,false,false,false,true,true
+//		#default,_result,,,,,
+//		,result,table,_time,_value,_field,location
+//		,,0,2020-02-18T10:34:08.135814545Z,23.4,temp,livingroom
+//		,,0,2020-02-18T22:08:44.850214724Z,21.6,temp,livingroom
+//
+//      #datatype,string,long,dateTime:RFC3339,double,string,string,string
```

Comment:
It's nice to match surrounding white space.

```suggestion
//		#datatype,string,long,dateTime:RFC3339,double,string,string,string
```